test/jdk/javax/swing/JTree/4518432/JTreeNodeCopyPasteTest.java

```diff
@@ -0,0 +1,237 @@
+ * questions.
+ */
+
+import java.awt.Dimension;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.awt.event.FocusAdapter;
+import java.awt.event.FocusEvent;
+import java.awt.event.InputEvent;
+import java.awt.event.KeyEvent;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+import javax.swing.JFrame;
+import javax.swing.JTree;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UIManager.LookAndFeelInfo;
+import javax.swing.UnsupportedLookAndFeelException;
+
+
+import static javax.swing.UIManager.getInstalledLookAndFeels;
+
+/*
+ * @test
+ * @key headful
+ * @bug 4518432
+ * @summary Verifies that Copying from JTree node and then changing the data in the component that was copied from,
+ *          is not causing the pastes to use the new edited data instead of the original copied data.
+ * @run main JTreeNodeCopyPasteTest
+ */
+public class JTreeNodeCopyPasteTest {
+
+    private static JFrame frame;
+    private static JTree tree;
+    private static Robot robot;
+    private static boolean isMac;
+    private static CountDownLatch treeGainedFocusLatch;
+
+    public static void main(String[] args) throws Exception {
+        runTest();
+    }
+
+    private static void runTest() throws Exception {
+        isMac = System.getProperty("os.name").toLowerCase().contains("os x");
+        robot = new Robot();
+        robot.setAutoDelay(100);
+        robot.setAutoWaitForIdle(true);
+
+        // Filter out Motif laf, as it doesn't support copy-paste in JTree.
+        List<String> lafs = Arrays.stream(getInstalledLookAndFeels())
+                                  .filter(laf -> !laf.getName().contains("Motif"))
+                                  .map(LookAndFeelInfo::getClassName)
+                                  .collect(Collectors.toList());
+        for (final String laf : lafs) {
+            treeGainedFocusLatch = new CountDownLatch(1);
+            try {
+                AtomicBoolean lafSetSuccess = new AtomicBoolean(false);
+                SwingUtilities.invokeAndWait(() -> {
+                    lafSetSuccess.set(setLookAndFeel(laf));
+                    if (lafSetSuccess.get()) {
+                        createUI();
+                    }
+                });
+                if (!lafSetSuccess.get()) {
+                    continue;
+                }
+
+                robot.waitForIdle();
+
+                // Waits until the tree gains focus
+                if (!treeGainedFocusLatch.await(3, TimeUnit.SECONDS)) {
+                    throw new RuntimeException("Test Failed, waited too long, " +
+                            "but the JTree can't gain focus for L&F: " + laf);
+                }
```

Comment:
The tree is the only component on the frame. By default, it will get focus. The call to `waitForIdle` guarantees all the events are processed. Even if tree doesn't get focus, the next thing you do is click a node in the tree, which brings the focus.